users_ldap_groups/models/res_users.py

```diff
@@ -1,27 +1,18 @@
-# -*- coding: utf-8 -*-
 # © 2018 Therp BV <https://therp.nl>
+# Copyright 2018 Brainbean Apps <https://brainbeanapps.com>
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
-from odoo import SUPERUSER_ID, api, models, registry
+from odoo import models
 
 
-class ResUsers(models.Model):
+class Users(models.Model):
     _inherit = 'res.users'
+    _description = 'User'
 
-    @classmethod
-    def _login(cls, db, login, password):
-        user_id = super(ResUsers, cls)._login(db, login, password)
-        if not user_id:
-            return user_id
-        with registry(db).cursor() as cr:
-            env = api.Environment(cr, SUPERUSER_ID, {})
-            user = env['res.users'].browse(user_id)
-            # check if this user came from ldap, rerun get_or_create_user in
-            # this case to apply ldap groups if necessary
-            ldaps = user.company_id.ldaps
-            if user.active and any(ldaps.mapped('only_ldap_groups')):
-                for conf in ldaps.get_ldap_dicts():
-                    entry = ldaps.authenticate(conf, login, password)
-                    if entry:
-                        ldaps.get_or_create_user(conf, login, entry)
-                        break
-        return user_id
+    def _check_credentials(self, password):
```

Comment:
this function is called for every request made to odoo, cf https://github.com/OCA/OCB/blob/12.0/odoo/addons/base/models/res_users.py#L610, https://github.com/OCA/OCB/blob/12.0/odoo/service/security.py#L14, https://github.com/OCA/OCB/blob/12.0/odoo/service/model.py#L35

you don't want to do this on every request. `_login` is just fine

Reply:
thanks for a hint, done